clipper_admin/clipper_admin/deployers/python.py

```diff
@@ -0,0 +1,134 @@
+from __future__ import print_function, with_statement, absolute_import
+
+import logging
+import shutil
+from ..version import __version__
+
+from .deployer_utils import save_python_function
+
+logger = logging.getLogger(__name__)
+
+
+def create_endpoint(
+        clipper_conn,
+        name,
+        input_type,
+        func,
+        default_output="None",
+        version=1,
+        slo_micros=3000000,
+        labels=None,
+        registry=None,
+        base_image="clipper/python-closure-container:{}".format(__version__),
+        num_replicas=1):
+    """Registers an app and deploys provided predict function as a model.
+
+    Parameters
+    ----------
+    name : str
+        The to be assigned to the registered app and deployed model.
+    predict_function : function
+        The prediction function. Any state associated with the function should be
+        captured via closure capture.
+    input_type : str
+        The input_type to be associated with the registered app and deployed model.
+        One of "integers", "floats", "doubles", "bytes", or "strings".
+    default_output : string, optional
+        The default prediction to use if the model does not return a prediction
+        by the end of the latency objective.
+    model_version : Any object with a string representation (with __str__ implementation), optional
+        The version to assign the deployed model.
+    slo_micros : int
+        The query latency objective for the application in microseconds.
+        This is the processing latency between Clipper receiving a request
+        and sending a response. It does not account for network latencies
+        before a request is received or after a response is sent.
+    labels : list of str, optional
+        A list of strings annotating the model.
+    num_containers : int, optional
+        The number of replicas of the model to create. More replicas can be
+        created later as well.
+    """
+
+    clipper_conn.register_application(name, input_type, default_output,
+                                      slo_micros)
+    deploy_python_closure(clipper_conn, name, version, input_type, func,
+                          base_image, labels, registry, num_replicas)
+
+    clipper_conn.link_model_to_app(name, name)
+
+
+def deploy_python_closure(
```

Comment:
After invoking `deploy_python_closure(conn, "m7", "1", "ints", func) ` with a connection backed by the docker container manager, I encountered an error with the following back trace:

```
Traceback (most recent call last):
  File "/Users/Corey/Documents/RISE/clipper/clipper_admin/clipper_admin/deployers/check_and_write_deps.py", line 127, in <module>
    conda_dep_fname, pip_dep_fname):
  File "/Users/Corey/Documents/RISE/clipper/clipper_admin/clipper_admin/deployers/check_and_write_deps.py", line 91, in check_solvability_write_deps
    missing_packages = missing_packages_error.pkgs
AttributeError: 'ResolvePackageNotFound' object has no attribute 'pkgs'

17-08-30:02:04:13 INFO     [python.py:127] Python closure saved
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/Corey/Documents/RISE/clipper/clipper_admin/clipper_admin/deployers/python.py", line 138, in deploy_python_closure
    force=True)
  File "/Users/Corey/Documents/RISE/clipper/clipper_admin/clipper_admin/clipper_admin.py", line 281, in build_and_deploy_model
    container_registry)
  File "/Users/Corey/Documents/RISE/clipper/clipper_admin/clipper_admin/clipper_admin.py", line 346, in build_model
    docker_file_path = os.path.join(model_data_path, "Dockerfile")
  File "/usr/local/bin/anaconda2/envs/clipper/lib/python2.7/posixpath.py", line 70, in join
    elif path == '' or path.endswith('/'):
AttributeError: 'bool' object has no attribute 'endswith'
```

Reply:
This issue looks like a problem introduced by the latest version of conda. We got a bug report about this in #269.

What version of conda are you on?

Reply:
Can you try rolling back to a 4.3.* version of conda and running again. This is definitely a bug that we should fix, but isn't a new one introduced by this PR and shouldn't block this getting merged.

Reply:
I'm on a 4.3* version of conda: 4.3.23

Reply:
Ah, that seems to be the latest version. Will roll back and test with something earlier